arduino/libraries/loader.go

```diff
@@ -186,3 +192,23 @@ func addExamplesToPathList(examplesPath *paths.Path, list *paths.PathList) error
 	}
 	return nil
 }
+
+// containsHeaderFile returns true if the directory contains a ".h" file.
+// Returns false otherwise
+func containsHeaderFile(d *paths.Path) (bool, error) {
+	if d.NotExist() {
+		// A directory that doesn't exist can't obviously contain any header file
+		return false, nil
+	}
```

Comment:
I'd remove this check because:
* we should never reach this point with `d` nonexistent
* if it will ever happen, it's better to fail with an error on `ReadDir` later